### PR TITLE
Catalogs

### PR DESCRIPTION
**Pull request by SunilSimha**
Adding Pan-STARRS to the list of surveys to check from. Also includes some minor bug fixes.

### CODE REVIEW

File: frb/surveys/panstarrs.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            url += "&{}={}".format(col,extension)
+    else:
+        urlbase = url + "&red="
+        url = []
+        for extensions in file_extensions:
+            url.append(urlbase+extensions)
+    return url
+ 
+def _check_columns(columns,table,release):
+    dcols = {}
```
Comment (profxj) [inline frb/surveys/panstarrs.py]: add a brief doc string

File: frb/surveys/panstarrs.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            hdu: fits header data unit for the downloaded image
+        """
+        assert len(filt)==1 and filt in "grizy", "Filter name must be one of 'g','r','i','z','y'"
+        url = _get_url(self.coord,imsize=imsize,filt=filt,imgformat='fits')[0]
+        imagedat = fits.open(astroutils.data.download_file(url,cache=True,show_progress=False,timeout=timeout))[0]
+        return imagedat
+
+
+
+def _get_url(coord,imsize=30*u.arcsec,filt="i",output_size=None,imgformat="fits",color=False):
```
Comment (profxj) [inline frb/surveys/panstarrs.py]: add a brief doc string

File: frb/surveys/panstarrs.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        dcols[col.lower()] = 1
+    badcols = []
+    for col in columns:
+        if col.lower().strip() not in dcols:
+            badcols.append(col)
+    if badcols:
+        raise ValueError('Some columns not found in table: {}'.format(', '.join(badcols)))
+
+def _check_legal(table,release):
+    """
```
Comment (profxj) [inline frb/surveys/panstarrs.py]: add parameters to the doc string

File: frb/surveys/sdss.py [hunk 4]
```diff
@@ -93,7 +94,7 @@ def get_catalog(self, photoobj_fields=None, timeout=None, print_query=False):
         query += "p.objid, "
 
         query += "pz.z as redshift, pz.zErr as redshift_error\n"
-        query += "FROM PhotoObjAll as p\n"
+        query += "FROM PhotoObj as p\n"
```
Comment (profxj) [inline frb/surveys/sdss.py]: Can you remind me of the difference between `PhotoObjAll` and `PhotoObj` ?

File: frb/galaxies/frbgalaxy.py [hunk 5]
```diff
@@ -274,27 +274,69 @@ def gen_cigale_data_in(self, ID=None, filename='data.fits', overwrite=False):
         # Convert DES and SDSS fluxes to mJy
         ABmagbands = ["DES_"+band for band in defs.DES_bands]
         ABmagbands += ["SDSS_"+band for band in defs.SDSS_bands]
+        ABmagbands += ['VLT_'+band for band in defs.VLT_bands]
+        #ABmagbands += ['VISTA_'+band for band in defs.VISTA_bands]
```
Comment (profxj) [inline frb/galaxies/frbgalaxy.py]: Is VISTA implemented or not?  I see it in `defs.py` but it is commented out here.
Either is fine, but it looks like it is a mixed implementation.

File: frb/surveys/sdss.py (hunk 4 above)
Comment (SunilSimha) [reply frb/surveys/sdss.py]: PhotoObjAll contains all detections and will often have multiple entries corresponding to the same object. PhotoObj is the cleaned version of this table without these redundancies.

File: frb/galaxies/frbgalaxy.py (hunk 5 above)
Comment (SunilSimha) [reply frb/galaxies/frbgalaxy.py]: It has been implemented below. I removed VISTA from the list of AB magnitude bands as their magnitudes are in the Vega system. There is a snippet below in the file that converts VITS mags to fluxes.

**Comment by profxj** [comment]
Merging.

**Pull request merged**